typescript/infra/scripts/verify-validator.ts

```diff
@@ -104,170 +89,259 @@ class S3Wrapper {
   }
 }
 
-async function main() {
-  const {
-    a: _validatorAddress,
-    p: prospectiveBucket,
-    c: controlBucket,
-  } = await getArgs();
+class Validator {
+  private readonly controlS3BucketClient: S3Wrapper;
+  private readonly prospectiveS3BucketClient: S3Wrapper;
 
-  const cClient = new S3Wrapper(controlBucket);
-  const pClient = new S3Wrapper(prospectiveBucket);
+  // accumulators for stats
+  private extraCheckpoints!: number[];
+  private missingCheckpoints!: number[];
+  private invalidCheckpoints!: number[];
+  private modTimeDeltasS!: number[];
+  private fullyCorrectCheckpoints!: number[];
+  private missingInARow!: number;
```

Comment:
nit: `consecutiveMissing`

Reply:
As @tkporter pointed out they are not necearrially consecutive indexes because the checkpoints can be done for multiple messages at a time. This is why I kept it as `missingInARow`. If you have a better name let me know, but I don't think we should use `consecutive` as I find it to be misleading.

Reply:
`inARow` implies the same thing as `consecutive` to me
imo neither name is good if the above is true